pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -425,6 +449,37 @@ public boolean index(GenericRow row, @Nullable RowMetadata rowMetadata) {
     return canTakeMore;
   }
 
+  private void updateMinMaxValue(GenericRow row) {
```

Comment:
Can this be done inline during `index()` (inside `addForwardIndex`)? Doing it separately may have several redundant operations, that may impact use cases with high ingestion rate.

Reply:
will do